File: baekjoon/10828/10828.py
# push, pop, size,empty, top
import sys
input = sys.stdin.readline

num = int(input())
stack = []
for i in range(num):
    command = input().split()
    if command[0] == "push":
        stack.append(command[1])
    elif command[0] == "pop":
        if not stack:
            print(-1)
        else:
            print(stack.pop())
    elif command[0] == "size":
        print(len(stack))
    elif command[0] == "top":
        if not stack:
            print(-1)
        else:
            print(stack[-1])
    elif command[0] == "empty":
        if not stack:
            print(1)
        else:
            print(0)

File: 10828.cpp
#include<iostream>
#include<stack>
#include<string>
using namespace std;

int main(){
    int n;
    string str;
    int m;
    stack<int>s;
    cin >> n;
    while (n--) {
        cin >> str;
        if (str == "push") {
            cin >> m;
            s.push(m);
        }
        else if (str == "pop") {
            if (s.size() == 0)
                cout << -1 << '\n';
            else {
                cout << s.top() << '\n';
                s.pop();
            }
        }
        else if (str == "size")
            cout << s.size() << '\n';
        else if (str == "empty")
            cout << s.empty() << '\n';
        else {
            if (s.size() == 0)
                cout << -1 << '\n';
            else
                cout << s.top() << '\n';
        }
    }
    return 0;
}
